### PR TITLE
Add new book image upload

### PR DESCRIPTION
**Pull request by fadhilmch**

### CODE REVIEW

**Review by bt-s — COMMENTED** [review]
Please disregard this comment.

File: src/components/AddNewBookForm/AddNewBookForm.js [hunk 1]
```diff
@@ -4,15 +4,19 @@ import {connect} from 'react-redux';
...
-  const {addNewUserBook, author, title, rating, firebase} = props;
+const AddNewBookFormBase = props => {
+  const {addNewUserBook, author, title, rating, firebase, files} = props;
   const [description, setDescription] = useState('');
   const [location, setLocation] = useState({
     lat: 0,
     lon: 0
   });
+  const [type, setType] = useState('lend');
+  var imageUrls = [];
```
Comment (bt-s) [inline src/components/AddNewBookForm/AddNewBookForm.js:18]: Maybe change this to ```let```

File: src/components/AddNewBookForm/AddNewBookForm.js [hunk 2]
```diff
@@ -33,40 +52,75 @@ const AddNewBookForm = props => {
...
-        imageURL: '',
+        imageUrls,
         type: 'lend',
         createdAt: new Date().getTime(),
         updatedAt: new Date().getTime()
       })
-      .then(() => {
-        addNewUserBook('success');
+      .then(res => {
+        console.log(res.id);
```
Comment (bt-s) [inline src/components/AddNewBookForm/AddNewBookForm.js:60]: Maybe remove this console.log statement before merging with master.

File: src/components/AddNewBookForm/AddNewBookForm.js [hunk 3]
```diff
@@ -33,40 +52,75 @@ const AddNewBookForm = props => {
...
+          files.map(file =>
+            uploadPictureToFirebase(
+              file,
+              `books/${res.id}`,
+              firebase,
+              handleImageUploaded
+            )
+          )
+        ).then(() => {
+          console.log(imageUrls);
```
Comment (bt-s) [inline src/components/AddNewBookForm/AddNewBookForm.js:71]: Idem

File: src/components/AddNewBookForm/AddNewBookForm.js [hunk 4]
```diff
@@ -33,40 +52,75 @@ const AddNewBookForm = props => {
...
+          console.log(res.id);
```
Comment (bt-s) [inline src/components/AddNewBookForm/AddNewBookForm.js:72]: Idem

File: src/components/AddNewBookForm/AddNewBookForm.js [hunk 5]
```diff
@@ -33,40 +52,75 @@ const AddNewBookForm = props => {
...
   };
 
+  const handleSubmit = () => {
+    addNewUserBook('loading');
+    console.warn('[ADD_NEW_BOOK] Calling API to add New Book');
+    storeData();
+  };
+
   useEffect(() => {
     navigator.geolocation.getCurrentPosition(parseLocation);
```
Comment (bt-s) [inline src/components/AddNewBookForm/AddNewBookForm.js:88]: Cool feature!

File: src/components/AddNewBookForm/_DragAndDrop.scss [hunk 6]
```diff
@@ -16,7 +17,26 @@
...
+}
+
+.input {
+  opacity: 0;
+  /* IE 8 */
+  -ms-filter: 'progid:DXImageTransform.Microsoft.Alpha(Opacity=0)';
+  /* IE 5-7 */
+  filter: alpha(opacity=0);
+  /* Netscape or and older firefox browsers */
+  -moz-opacity: 0;
```
Comment (bt-s) [inline src/components/AddNewBookForm/_DragAndDrop.scss:37]: I think that Babel takes care of this already.

File: src/pages/AddNewBookPage.js [hunk 7]
```diff
@@ -27,42 +33,31 @@ const AddNewBook = props => {
...
+        <div className="text">{file.name}</div>
+        <button onClick={() => removeFiles(i)}>
+          <FontAwesomeIcon icon={faTimesCircle} />
         </button>
+        <div id="gradient" />
       </div>
     );
   };
 
+  useEffect(() => {
```
Comment (bt-s) [inline src/pages/AddNewBookPage.js:46]: It seems like this useEffect is not doing anything (yet).

File: src/pages/_AddNewBookPage.scss [hunk 8]
```diff
@@ -0,0 +1,93 @@
+.image-box img {
+  position: absolute;
+  left: 50%;
+  top: 50%;
+  height: 100%;
+  width: auto;
+  -webkit-transform: translate(-50%, -50%);
```
Comment (bt-s) [inline src/pages/_AddNewBookPage.scss:7]: Babel takes care of this.

File: src/pages/_AddNewBookPage.scss [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    bottom,
+    rgba(255, 255, 255, 0) 60%,
+    rgba(255, 255, 255, 0.81) 95%
+  );
+  background: -ms-linear-gradient(
+    bottom,
+    rgba(255, 255, 255, 0) 60%,
+    rgba(255, 255, 255, 0.81) 95%
+  );
+  background: linear-gradient(
```
Comment (bt-s) [inline src/pages/_AddNewBookPage.scss:47]: You will only have to keep this one, since we have an autoprefixer.

**Review by bt-s — COMMENTED** [review]
Overall, this looks really good Fadhil. Good job!

I have observed the following issues:
- When trying to publish a book (locally) I get the following error ```TypeError: JSON.parse(...) is null[Learn More]```
- There are some conflicts with the master branch
- There were some linting warnings, which I have fixed in a commit

File: src/components/AddNewBookForm/AddNewBookForm.js (hunk 1 above)
Comment (Envl) [reply src/components/AddNewBookForm/AddNewBookForm.js:18]: No need to change I think. `const` only affects the `List` itself, not objects in it.

File: src/components/BookItem/BookItem.js [hunk 10]
```diff
@@ -4,7 +4,6 @@ import {Link} from 'react-router-dom';
 
 import reactJPG from '../../images/fullstackreact.jpg';
 import reactSVG from '../../images/react.svg';
-import * as ROUTES from '../../constants/routes';
```
Comment (Envl) [inline src/components/BookItem/BookItem.js]: There would be a conflict about importing 2 .scss files. Just keep the origin one. Actually there is nothing added in this PR, so I don't know why that could be a conflict.

**Review by Envl — APPROVED** [review]
I think it's ok to merge. Just resolve these small conflicts.

**Pull request merged**